goaviatrix/firewall.go

```diff
@@ -164,3 +169,81 @@ func PolicyToMap(p *Policy) map[string]interface{} {
 		"description": p.Description,
 	}
 }
+
+func (c *Client) AddFirewallPolicy(fw *Firewall) error {
+	action := "append_stateful_firewall_rules"
+
+	rules, err := json.Marshal(fw.PolicyList)
+	if err != nil {
+		return fmt.Errorf("could not marshal firewall policies: %v", err)
+	}
+
+	return c.PostAPI(action, struct {
+		Action string `form:"action"`
+		CID    string `form:"CID"`
+		GwName string `form:"gateway_name"`
+		Rules  string `form:"rules"`
+	}{
+		Action: action,
+		CID:    c.CID,
+		GwName: fw.GwName,
+		Rules:  string(rules),
+	}, BasicCheck)
+}
+
+func (c *Client) DeleteFirewallPolicy(fw *Firewall) error {
+	action := "delete_stateful_firewall_rules"
+
+	rules, err := json.Marshal(fw.PolicyList)
+	if err != nil {
+		return fmt.Errorf("could not marshal firewall policies: %v", err)
+	}
+
+	return c.PostAPI(action, struct {
+		Action string `form:"action"`
+		CID    string `form:"CID"`
+		GwName string `form:"gateway_name"`
+		Rules  string `form:"rules"`
+	}{
+		Action: action,
+		CID:    c.CID,
+		GwName: fw.GwName,
+		Rules:  string(rules),
+	}, func(action string, reason string, ret bool) error {
+		if !ret {
+			// Tried to delete a rule that did not exist, we don't need to fail the apply.
+			if strings.Contains(reason, "Empty rules in Stateful Firewall to delete") {
+				return nil
```

Comment:
Similar comment to one in the other PR .You could ignore the error where DeleteFirewallPolicy is being called